viscy/light/engine.py

```diff
@@ -723,12 +728,18 @@ def validation_step(
             self.validation_step_outputs.extend(
                 _detach_sample((anchor, pos_img, neg_img), self.log_samples_per_batch)
             )
+        self.validation_losses.append(loss)
         return loss
 
     def on_validation_epoch_end(self) -> None:
         super().on_validation_epoch_end()
+        val_loss_epoch = torch.stack(self.validation_losses).mean()
+        self.log(
+            "val/loss_epoch", val_loss_epoch, prog_bar=True, logger=True, sync_dist=True
+        )
         self._log_samples("val_samples", self.validation_step_outputs)
         self.validation_step_outputs = []
+        self.validation_losses = []
```

Comment:
Logging loss per epoch is already done automatically. There is no need to do this manual aggregation.